app/controllers/notifications_controller.rb

```diff
@@ -66,4 +67,12 @@ def if_not_signed_in
         end
       end
     end
+
+    # Stops here if the signed in user is not the owner of
+    # the notification set by :set_notification.
+    def ensure_user_owns_notification!
+      unless @notification.user == current_user
```

Comment:
@tcdowney @julianguyen Well, I didn't want to break the existing pattern, so I thought, it would be better not to touch `set_notification`. And, as I look at it again, I guess it might be worth thinking about a "global solution" (by that I mean a pattern to be followed in all controllers) to check whether the user is owner of the object, since we can find this problem in other controllers, too (take medications#destroy and medications#update as an example). I'm pretty new to contributing to open source, but maybe it's best if I open another issue about that!? Then you could merge your work in [#319](https://github.com/julianguyen/ifme/pull/319) instead of mine here, since you not only addressed this problem but also wrote tests for the other actions, and then think about away to implement/refactor "permission checking" in all controllers in a consistent way.